sdjournal/journal.go

```diff
@@ -293,6 +293,23 @@ func (j *Journal) GetRealtimeUsec() (uint64, error) {
 	return uint64(usec), nil
 }
 
+// GetCursor gets the cursor of the current journal entry.
+func (j *Journal) GetCursor() (string, error) {
+	var d *C.char
+
+	j.mu.Lock()
+	r := C.sd_journal_get_cursor(j.cjournal, &d)
+	j.mu.Unlock()
+
+	if r < 0 {
+		return "", fmt.Errorf("failed to get cursor: %d", r)
```

Comment:
I think you can wrap an (inverted) `r` into a `syscall.Errno` and expose a more meaningful error to users.